internal/wire/analyze.go

```diff
@@ -398,8 +447,8 @@ func verifyAcyclic(providerMap *typeutil.Map, hasher typeutil.Hasher) []error {
 				continue
 			}
 			pt := x.(*ProvidedType)
-			if pt.IsValue() {
-				// Leaf: values do not have dependencies.
+			if pt.IsValue() || pt.IsField() {
+				// Leaf: values or fields do not have dependencies.
```

Comment:
But fields do have a dependency on the parent struct. I think the test case that would break here would be:

```go
type S struct {
  Foo *S
}

var set = wire.NewSet(wire.FieldsOf(new(*S), "Foo"))
```

Reply:
Yes, added check and test.

Reply:
Comment stale now.

Reply:
Ping, please remove comment.